File: 1984-minimum-difference-between-highest-and-lowest-of-k-scores/1984-minimum-difference-between-highest-and-lowest-of-k-scores.java
class Solution {
    public int minimumDifference(int[] nums, int k) {
        Arrays.sort(nums);
        var n = nums.length;
        
        var ans = nums[n-1] - nums[0];
        if(k == n) return ans;
        for(var i=k; i<=n; i++)
            ans = Math.min(ans, nums[i-1] - nums[i-k]);
        
        return ans;
    }
}

File: 1984-minimum-difference-between-highest-and-lowest-of-k-scores/1984-minimum-difference-between-highest-and-lowest-of-k-scores.cs
public class Solution {
    public int MinimumDifference(int[] nums, int k) {
        Array.Sort(nums);
        var n = nums.Length;
        
        var ans = nums[n-1] - nums[0];
        if(k == n) return ans;
        for(var i=k; i<=n; i++)
            ans = Math.Min(ans, nums[i-1] - nums[i-k]);
        
        return ans;
    }
}